guides/v2.3/graphql/functional-testing.md

```diff
@@ -170,3 +170,126 @@ $registry->register('isSecureArea', false);
+{
+    ...
+    
+    self::expectExceptionMessage("Expected exception message goes here...");
+    
+    ...
+}
+
+```
+or 
+```php
+public function testMyExceptionTest()
+{
+    ...
+    
+    $this->expectExceptionMessage("Expected exception message goes here...");
+    
+    ...
+}
+```
+{% include note.html type="info" content="Please note that the exception message expectation should be defined before invoking a logic that generates the exception." %}
+
+For example, if `customer A` wants to retrieve information about `customer B cart. In this situation, `customer A` will get an error - "`The current user cannot perform operations on cart "XXXXX`", where `XXXXX` - is an unique ID of `customer B` cart. See code below to understand how to cover this example with `expectExceptionMessage` function:
+
+```php
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_items_saved.php
+     */
+    public function testGetCartFromAnotherCustomer()
+    {
+        $reservedOrderId = 'test_order_item_with_items';
+        $this->quoteResource->load(
+            $this->quote,
+            $reservedOrderId,
+            'reserved_order_id'
+        );
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
+        $query = $this->prepareGetCartQuery($maskedQuoteId);
+        self::expectExceptionMessage("The current user cannot perform operations on cart \"$maskedQuoteId\"");
+        $this->graphQlQuery($query);
+    } 
+```
+
+#### Define expected exception message in test function annotation
+As an alternative for calling `expectExceptionMessage` method use a predefined directive `@expectedExceptionMessage`:
+
+```php
+    /**
+     * @expectedException \Exception
+     * @expectedExceptionMessage Expected exception message goes here...
+     */
+```
+
+Example: a customer wants to retrieve information about his own cart but he provides incorrect cart ID.
+
+**Query**
+```text
+{
+  cart(cart_id: "YYYYY") {
+    items {
+      __typename
+      id
+      qty
+    }
+  }
+}
+```
+
+**Result**
+```json
+{
+  "errors": [
+    {
+      "message": "Could not find a cart with ID \"YYYYY\"",
+      "category": "graphql-no-such-entity",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": [
+        "cart"
+      ]
+    }
+  ],
+  "data": {
+    "cart": null
+  }
+}
+```
+
+See the code below how to cover this example with `@expectExceptionMessage` annotation:
+```php
+    /**
+     * @expectedException \Exception
+     * @expectedExceptionMessage Could not find a cart with ID "non_existent_masked_id"
+     */
+    public function testGetNonExistentCart()
+    {
+        $maskedQuoteId = 'non_existent_masked_id';
+        $query = $this->prepareGetCartQuery($maskedQuoteId);
+        
+        $this->graphQlQuery($query);
+    }
+```
+
+There is a list of functions which you can use to cover expected exceptions:  
+- expectException
```

Comment:
Wrap the functions with a text code block

Reply:
> Wrap the functions with a text code block

**done**

Thank you, @keharper 👍